src/command.ts

```diff
@@ -0,0 +1,236 @@
+    locations = filesArgs;
+  });
+
+// TODO examples and formatting
+program.on("--help", () => {
+  console.log("");
+  console.log("Examples:");
+  console.log("  $ dolos -l javascript *.js");
+  console.log(
+    "Gives dolos all the files in the current directory and tells that tells dolos that they are in javascript",
+  );
+  console.log("");
+  console.log("Specifies the gap size.");
+  console.log("  $ dolos *.js -g 0");
+  console.log("  [[[0, 2], [9, 11]], [[4, 5], [13, 14]]]");
+  console.log("  $ dolos *.js -g 1");
+  console.log("  [[[0, 5], [9, 14]]]");
+});
+
+program.parse(process.argv);
+/**
+ * This function allows one to merge two maps into one.
+ * @param matchesPerFile A map containing the matches. This map will be modified in place.
+ * @param newMatches The map you want to merge into the first map.
+ */
+function compose(
+  matchesPerFile: Map<string, Matches<number>>,
+  newMatches: Map<string, Matches<number>>,
+) {
+  newMatches.forEach((matches, matchedFileName) => {
+    matches.forEach((matchLocations, matchingFile) => {
+      let map: Matches<number> | undefined = matchesPerFile.get(matchingFile);
+      if (map === undefined) {
+        map = new Map();
+        matchesPerFile.set(matchedFileName, map);
+      }
+      map.set(matchingFile, matchLocations);
+    });
+  });
+}
+
+/**
+ * Groups files per directory.
+ * @param locations The locations you want to group.
+ * @returns An array where each subArray contains files that are in the same directory.
+ */
+function groupPerDirectory(files: string[]): string[][] {
+  const locationsFragments = files.map(filePath => filePath.split(path.sep));
+  const filesGroupedPerDirectoryMap: Map<string, string[]> = new Map();
+
+  let baseDirIndex = 0;
+  let baseDir = locationsFragments[0][0];
+  while (
+    locationsFragments.every(filePathFragments => filePathFragments[baseDirIndex] === baseDir)
+  ) {
+    baseDirIndex += 1;
+    baseDir = locationsFragments[0][baseDirIndex];
+  }
+
+  locationsFragments.forEach(filePathFragments => {
+    let groupedFiles: string[] | undefined = filesGroupedPerDirectoryMap.get(
+      filePathFragments[baseDirIndex],
+    );
+    if (groupedFiles === undefined) {
+      groupedFiles = new Array();
+      filesGroupedPerDirectoryMap.set(filePathFragments[baseDirIndex], groupedFiles);
+    }
+    groupedFiles.push(path.join(...filePathFragments));
+  });
+
+  return [...filesGroupedPerDirectoryMap.values()];
+}
+
+(async () => {
+  let baseFiles: string[] | undefined;
+  let groupAmount: number;
+  const tokenizer = new CodeTokenizer(program.language);
+  const results: Map<string, Matches<number>> = new Map();
+  let baseFileMatches: Map<string, Matches<number>> = new Map();
+
+  // if the -d and the -b flag are active, filter out all the basefile locations
+  if (program.directory && program.base) {
+    program.base = path.normalize(program.base);
+
+    baseFiles = locations.filter(filePath => filePath.startsWith(program.base));
```

Comment:
No there is not, but when both the -d and the -b options are used at the same time, the -b options only specifies the directory where the base files are located. The actual files themself are given with the rest of the files. I did it it this way because commander cannot handle two variable arguments at the same time.